lib/dagDisplay.ml

```diff
@@ -25,42 +25,57 @@ let image_normal_txt conf base p fname width height =
       (if width = 0 then "" else "max-width:" ^ string_of_int width ^ "px;")
       (if height = 0 then "" else "max-height:" ^ string_of_int height ^ "px;")
   in
-  Format.sprintf {|<a href="%sm=IM&%s&k=%s">%s</a>|}
-    (commd conf : Adef.escaped_string :> string)
-    (acces conf base p : Adef.escaped_string :> string)
-    k r
+  (if p_getenv conf.env "cgl" = Some "on" then r
+  else
+    Format.sprintf {|<a href="%sm=IM&%s&k=%s">%s</a>|}
+      (commd conf : Adef.escaped_string :> string)
+      (acces conf base p : Adef.escaped_string :> string)
+      k r)
   |> Adef.safe
 
 let image_url_txt conf url_p url height : Adef.safe_string =
   let image_txt = Utf8.capitalize_fst (transl_nth conf "image/images" 0) in
+  let hed, trl =
+    if p_getenv conf.env "cgl" = Some "on" then ("", "")
+    else
+      ( Format.sprintf {|<a href="%s">|} (url_p : Adef.escaped_string :> string),
+        "</a>" )
+  in
   Format.sprintf
-    {|<a href="%s"><img src="%s" alt="%s" title="%s"
-      style="%s"></a>|}
-    (url_p : Adef.escaped_string :> string)
+    {|%s<img src="%s" alt="%s" title="%s"
+      style="%s">%s|}
+    hed
     (url : Adef.escaped_string :> string)
     image_txt image_txt
     (if height = 0 then "" else "max-height:" ^ string_of_int height ^ "px;")
+    trl
   |> Adef.safe
 
 let image_url_txt_with_size conf url_p url width height : Adef.safe_string =
   let image_txt = Utf8.capitalize_fst (transl_nth conf "image/images" 0) in
+  let hed, trl =
+    if p_getenv conf.env "cgl" = Some "on" then ("", "")
+    else
+      ( Format.sprintf {|<a href="%s">|} (url_p : Adef.escaped_string :> string),
+        "</a>" )
+  in
   Format.sprintf
-    {|<a href="%s"><img src="%s"%s%s alt="%s" title="%s"
-      style="%s %s">%s</a>|}
-    (url_p : Adef.escaped_string :> string)
+    {|%s<img src="%s"%s%s alt="%s" title="%s"
+      style="%s %s">%s%s|}
+    hed
     (url : Adef.escaped_string :> string)
     (if width = 0 then "" else " width=" ^ string_of_int width)
     (if height = 0 then "" else " height=" ^ string_of_int height)
     image_txt image_txt
     (if width = 0 then "" else "max-width:" ^ string_of_int width ^ "px;")
     (if height = 0 then "" else "max-height:" ^ string_of_int height ^ "px;")
-    image_txt
+    image_txt trl
```

Comment:
not from this PR, but do you know why `image_url_txt_with_size` print a `image_txt` after the `<img ...>` but not `image_url_txt`?

Reply:
I did a little clean up of `image_url_txt` in this commit (untested): https://github.com/geneweb/geneweb/commit/fc22cea02467f4addadf1a75eef8065649c35ac4